modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java

```diff
@@ -399,4 +405,49 @@ static CatalogIndexDescriptor indexOrThrow(Catalog catalog, int indexId) throws
 
         return index;
     }
+
+    /**
+     * Collects all table indexes (including dropped) that the table had in the requested catalog version range.
+     *
+     * <p>It is expected that at least one version of the catalog contains table indexes.</p>
+     *
+     * @param catalogService Catalog service.
+     * @param tableId Table ID for which indexes will be collected.
+     * @param catalogVersionFrom Catalog version from which indexes will be collected (including).
+     * @param catalogVersionTo Catalog version up to which indexes will be collected (including).
+     * @return Table indexes sorted by {@link CatalogIndexDescriptor#id()}.
+     */
+    public static List<CatalogIndexDescriptor> collectIndexes(
+            CatalogService catalogService,
+            int tableId,
+            int catalogVersionFrom,
+            int catalogVersionTo
+    ) {
+        assert catalogVersionFrom <= catalogVersionTo : "from=" + catalogVersionFrom + ", to=" + catalogVersionTo;
+
+        if (catalogVersionFrom == catalogVersionTo) {
+            List<CatalogIndexDescriptor> indexes = catalogService.indexes(catalogVersionFrom, tableId);
+
+            assert !indexes.isEmpty() : "catalogVersion=" + catalogVersionFrom + ", tableId=" + tableId;
+
+            return indexes;
+        }
+
+        var indexByIdMap = new Int2ObjectOpenHashMap<CatalogIndexDescriptor>();
```

Comment:
Yes, I can use it, but I think my current approach will be faster.

Reply:
I'm not talking about efficiency (it's nearly the same), it would be simply shorter in terms of code. If you can remove the sorting, then my suggestion is irrelevant.

Reply:
fix it